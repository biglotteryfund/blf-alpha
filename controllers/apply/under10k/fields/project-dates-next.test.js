/* eslint-env jest */
'use strict';
const {
    fieldProjectStartDateCheck,
    fieldProjectEndDate,
} = require('./project-dates-next');

describe('fieldProjectStartDateCheck', function () {
    test('both options enabled by default', function () {
        const field = fieldProjectStartDateCheck('en');
        expect(field.options).toStrictEqual([
            { label: expect.any(String), value: 'asap' },
            { label: expect.any(String), value: 'exact-date' },
        ]);
    });

    test('exact date disabled in England', function () {
        const field = fieldProjectStartDateCheck('en', {
            projectCountry: 'england',
        });
        expect(field.options).toStrictEqual([
            {
                value: 'asap',
                label: expect.any(String),
            },
            {
                value: 'exact-date',
                label: expect.any(String),
                attributes: { disabled: 'disabled' },
            },
        ]);
    });

    test('exact date disabled outside England when project is responding to COVID-19', function () {
        const field = fieldProjectStartDateCheck('en', {
            projectCountry: 'scotland',
            supportingCOVID19: 'yes',
        });
        expect(field.options).toStrictEqual([
            {
                value: 'asap',
                label: expect.any(String),
            },
            {
                value: 'exact-date',
                label: expect.any(String),
                attributes: { disabled: 'disabled' },
            },
        ]);
    });
});

describe('fieldProjectEndDate', function () {
    test('guidance states 6 months in England', function () {
        const field = fieldProjectEndDate('en', {
            projectCountry: 'england',
        });
        expect(field.explanation).toMatchSnapshot();
    });

    test('guidance states 12 months outside England', function () {
        const field = fieldProjectEndDate('en', {
            projectCountry: 'scotland',
        });
        expect(field.explanation).toMatchSnapshot();
    });

    test('guidance states 12 months when no country is selected', function () {
        const field = fieldProjectEndDate('en');
        expect(field.explanation).toMatchSnapshot();
    });
});
