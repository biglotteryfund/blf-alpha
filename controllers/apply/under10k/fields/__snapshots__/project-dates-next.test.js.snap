// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fieldProjectEndDate guidance states 6 months in England 1`] = `undefined`;

exports[`fieldProjectEndDate guidance states 12 months outside England 1`] = `"You have up to 12 months after award to spend the money. It can even be as short as just one day"`;

exports[`fieldProjectEndDate guidance states 12 months when no country is selected 1`] = `"You have up to 12 months after award to spend the money. It can even be as short as just one day"`;
